In [1]:
import numpy as np
import pandas as pd
import pydicom as dicom
import matplotlib.pylab as plt
import glob
import SimpleITK as sitk
import re

import os
pd.set_option('display.max_colwidth', None)
pd.set_option('display.max_columns', None)
dicom.config.convert_wrong_length_to_UN = True

In [2]:
#path = "F:/U-CAN-Lymfom_A/Ashish_SampleResampling/sample_data/"

#path = "F:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB"

#source_path = "/media/andres/T7 Shield/ucan_lymfom"
source_path = "E:/U-CAN-Lymfom_A"

destination_path = os.path.join(source_path, "metadata.xlsx")
selection_dataframe = os.path.join(source_path, "Final_Selected_exams_from_U-CAN-Lymfom.xlsx" )

In [3]:
directory_list = list()
for root, dirs, files in os.walk(source_path, topdown=False):
    for name in dirs:
        directory_list.append(os.path.join(root, name))

In [4]:
remove_list = [ 'PR----BONE-PULM-mm',
                'PR----Lunga-0.6-ax-mm',
                'PR----WB-Venfas-0.6-ax-mm',
                'PR----LUNG-1.25-AX-mm',
                'PR----WB-Ben-lunga-0.6-ax-mm',
                'PR----WB-Venfas-3-ax-mm',
                'PR----LUNG-1.25-AX-mm',
                'PR----BONE-1.25-AX-mm',
                'PR----LUNG-1.25-AX-mm',
                'PR----Lunga-0.6-ax-mm',
                'PR----SAVED-IMAGES-PR-mm',
                'PR----e1-QCFX-S-400-Static-mm',
                'PR----WB-Venfas-0.6-ax-mm',
                'PR----WB-VEN-AX-mm',
                'PR----WB-Ben-lunga-0.6-ax-mm',
                'PR----LUNG-1.25-AX-mm',
                'PR----THORAX-AX-mm',
                'PR----LUNG-1.25-AX-mm',
                'PR----THORAX-INANDAD-mm',
                'PR----KEY_IMAGES-PR-mm',
                'PR----SAVED-PR-mm',
                'Examinations that miss either CT or PET or both',
                'MR-',
                'sag',
                'cor',
                'ot-'
            ]

keep_list = ["CT-", "PT-"]

In [5]:
findir_lst = []
rejection_lst = []

for dir in directory_list:
    dir = dir.replace('\\', '/')
    if any(item.lower() in dir.lower() for item in keep_list) and all(
            item.lower() not in dir.lower() for item in remove_list):
        print(dir)
        findir_lst.append(dir)
    else:
        rejection_lst.append(dir)

E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm
E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_
E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-130439-6.3-UAS-WB-FDG-3D-CT-VENFAS-LUNG-1.25-AX-1.250000mm
E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.976562mm
E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-130439-6.3-UAS-WB-FDG-3D-CT-VENFAS-BONE-1.25-AX-1.250000mm_JK_
E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/PT-20160805-124458-UAS-WB-FDG-3D-CT-VENFAS-WB-FD

In [6]:
len(rejection_lst)

5207

In [7]:
ucan_md_dict = {'dir': [],
                'source_dir': [],
                'patient_dir': [],
                'image_dir': [],
                'dicom_img': [],
                'patient_id': [],
                'patient_age': [],
                'patient_sex': [],
                'patient_weight': [],
                'patient_size': [],
                'rows': [],
                'columns': [],
                'num_slices': [],
                'pixel_spacing': [],
                'slice_thickness': [],
                'img_pos': [],
                'img_orient': [],
                'for_uid': [],
                'att_corr': [],
                'recons_method': [],
                'image_type': [],
                'aquisition_dt': [],
                'aquisition_time': [],
                'study_desc': [],
                'series_desc': [],
                'protocol': [],
                'corr_img': [],
                'modality': [],
                'manufacturer': [],
                'manufacturer_model': [],
                'radiopharmaceutical': [],
                'radiopharmaceutical_volume': [],
                'radiopharmaceutical_start_time': [],
                'radionuclide_total_dose': [],
                'radionuclide_half_life': [],
                'radionuclide_positron_fraction': [],
                'radiopharmaceutical_start_date_time': [],
                }

#emptydir_lst = []

for dir in findir_lst:
    #if len(os.listdir(dir))<100:
    #    emptydir_lst.append(dir)
    #    continue

    ucan_md_dict['dir'].append(dir)
    ucan_md_dict['dicom_img'].append(os.listdir(dir)[0])
    
    ucan_md_dict['source_dir'].append(dir.rsplit(sep='/', maxsplit=2)[-3])
    ucan_md_dict['patient_dir'].append(dir.rsplit(sep='/', maxsplit=2)[-2])
    ucan_md_dict['image_dir'].append(dir.rsplit(sep='/', maxsplit=2)[-1])  
                                            
    #print(os.listdir(dir)[0])
    ds = dicom.dcmread(dir + '/' + str(os.listdir(dir)[0]))

    meta_info = str(ds.fix_meta_info)#.split('\n')
    
    max_range = 500
    
    #PatientID
    try:
        id = re.search(r"\b(0010, 0020)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['patient_id'].append(element)
    except:
        ucan_md_dict['patient_id'].append('')
    
    #PatientAge 
    try:
        id = re.search(r"\b(0010, 1010)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['patient_age'].append(element)
    except:
        ucan_md_dict['patient_age'].append('')

    #PatientSex
    try:
        id = re.search(r"\b(0010, 0040)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['patient_sex'].append(element)
    except:
        ucan_md_dict['patient_sex'].append('')

    #Patient Size
    try:
        id = re.search(r"\b(0010, 1020)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['patient_size'].append(element)
    except:
        ucan_md_dict['patient_size'].append('')

    #Patient Weight
    try:
        id = re.search(r"\b(0010, 1030)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['patient_weight'].append(element)
    except:
        ucan_md_dict['patient_weight'].append('')
    
    #Rows
    try:
        id = re.search(r"\b(0028, 0010)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['rows'].append(element)
    except:
        ucan_md_dict['rows'].append('')

    #Columns
    try:
        id = re.search(r"\b(0028, 0011)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['columns'].append(element)
    except:
        ucan_md_dict['columns'].append('')

    #Num of slices
    try:
        id = re.search(r"\b(07a1, 1002)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['num_slices'].append(element)
    except:
        ucan_md_dict['num_slices'].append('')

    #Pixel spacing
    try:
        id = re.search(r"\b(0028, 0030)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['pixel_spacing'].append(element)
    except:
        ucan_md_dict['pixel_spacing'].append('')

    #Slice thickness
    try:
        id = re.search(r"\b(0018, 0050)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['slice_thickness'].append(element)
    except:
        ucan_md_dict['slice_thickness'].append('')

    #Image positioning
    try:
        id = re.search(r"\b(0020, 0032)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['img_pos'].append(element)
    except:
        ucan_md_dict['img_pos'].append('')

    #Image orientation
    try:
        id = re.search(r"\b(0020, 0037)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['img_orient'].append(element)
    except:
        ucan_md_dict['img_orient'].append('')

    #Frame of reference UID
    try:
        id = re.search(r"\b(0020, 0052)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['for_uid'].append(element)
    except:
        ucan_md_dict['for_uid'].append('')

    #Attenuation Correction Method
    try:
        id = re.search(r"\b(0054, 1101)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['att_corr'].append(element)
    except:
        ucan_md_dict['att_corr'].append('')

    #Reconstruction Method
    try:
        id = re.search(r"\b(0054, 1103)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['recons_method'].append(element)
    except:
        ucan_md_dict['recons_method'].append('')

    #Image Type
    try:
        id = re.search(r"\b(0008, 0008)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['image_type'].append(element)
    except:
        ucan_md_dict['image_type'].append('')

    #Aquisition Date
    try:
        id = re.search(r"\b(0008, 0022)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['aquisition_dt'].append(element)
    except:
        ucan_md_dict['aquisition_dt'].append('')
    
    #Aquisition Time
    try:
        id = re.search(r"\b(0008, 0032)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['aquisition_time'].append(element)
    except:
        ucan_md_dict['aquisition_time'].append('')

    #Study Description
    try:
        id = re.search(r"\b(0008, 1030)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['study_desc'].append(element)
    except:
        ucan_md_dict['study_desc'].append('')

    #Series Description
    try:
        id = re.search(r"\b(0008, 103e)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['series_desc'].append(element)
    except:
        ucan_md_dict['series_desc'].append('')

    #Protocol Name 
    try:
        id = re.search(r"\b(0018, 1030)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['protocol'].append(element)
    except:
        ucan_md_dict['protocol'].append('')

    #Corrected Image
    try:
        id = re.search(r"\b(0028, 0051)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['corr_img'].append(element)
    except:
        ucan_md_dict['corr_img'].append('')

    #Modality
    try:
        id = re.search(r"\b(0008, 0060)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['modality'].append(element)
    except:
        ucan_md_dict['modality'].append('')

    #Manufacturer
    try:
        id = re.search(r"\b(0008, 0070)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['manufacturer'].append(element)
    except:
        ucan_md_dict['manufacturer'].append('')

    #Manufacturer_model
    try:
        id = re.search(r"\b(0008, 1090)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['manufacturer_model'].append(element)
    except:
        ucan_md_dict['manufacturer_model'].append('')
    # Radiopharmaceutical
    try:
        id = re.search(r"\b(0018, 0031)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['radiopharmaceutical'].append(element)
    except:
        ucan_md_dict['radiopharmaceutical'].append('')
     # Radiopharmaceutical Volume
    try:
        id = re.search(r"\b(0018, 1071)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['radiopharmaceutical_volume'].append(element)
    except:
        ucan_md_dict['radiopharmaceutical_volume'].append('')
     # Radiopharmaceutical Start Time
    try:
        id = re.search(r"\b(0018, 1072)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['radiopharmaceutical_start_time'].append(element)
    except:
        ucan_md_dict['radiopharmaceutical_start_time'].append('')
     # Radionuclide Total Dose
    try:
        id = re.search(r"\b(0018, 1074)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['radionuclide_total_dose'].append(element)
    except:
        ucan_md_dict['radionuclide_total_dose'].append('')
     # Radionuclide Half Life
    try:
        id = re.search(r"\b(0018, 1075)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['radionuclide_half_life'].append(element)
    except:
        ucan_md_dict['radionuclide_half_life'].append('')
     # Radionuclide Positron Fraction
    try:
        id = re.search(r"\b(0018, 1076)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['radionuclide_positron_fraction'].append(element)
    except:
        ucan_md_dict['radionuclide_positron_fraction'].append('')
     # Radiopharmaceutical Start Date Time
    try:
        id = re.search(r"\b(0018, 1078)\b", meta_info).start()
        element = meta_info[id:id+max_range].split('\n')[0].split(':')[1]
        ucan_md_dict['radiopharmaceutical_start_date_time'].append(element)
    except:
        ucan_md_dict['radiopharmaceutical_start_date_time'].append('')


c:\Users\asch4890\AppData\Local\anaconda3\envs\dlfia\lib\site-packages\pydicom\dataelem.py:868: UserWarning: Expected total bytes to be an even multiple of bytes per value. Instead received b'\x00\x00' with length 2 and struct format 'l' which corresponds to bytes per value of 4. This occurred while trying to parse (0009, 109d) according to VR 'SL'. Setting VR to 'UN'.
  warnings.warn(f"{message} Setting VR to 'UN'.")
c:\Users\asch4890\AppData\Local\anaconda3\envs\dlfia\lib\site-packages\pydicom\dataelem.py:868: UserWarning: Expected total bytes to be an even multiple of bytes per value. Instead received b'\x00\x00' with length 2 and struct format 'l' which corresponds to bytes per value of 4. This occurred while trying to parse (0009, 10b5) according to VR 'SL'. Setting VR to 'UN'.
  warnings.warn(f"{message} Setting VR to 'UN'.")
c:\Users\asch4890\AppData\Local\anaconda3\envs\dlfia\lib\site-packages\pydicom\dataelem.py:868: UserWarning: Expected total bytes to be an even multiple of 

In [8]:
"""
with open(r'F:/U-CAN-Lymfom_A/Ashish_SampleResampling/metadata_extraction/dir_lessthan_100images.txt', 'w') as fp:
    for item in emptydir_lst:
        # write each item on a new line
        fp.write("%s\n" % item)
    print('Done')

len(emptydir_lst)

Done
2399
"""

'\nwith open(r\'F:/U-CAN-Lymfom_A/Ashish_SampleResampling/metadata_extraction/dir_lessthan_100images.txt\', \'w\') as fp:\n    for item in emptydir_lst:\n        # write each item on a new line\n        fp.write("%s\n" % item)\n    print(\'Done\')\n\nlen(emptydir_lst)\n\nDone\n2399\n'

In [9]:
ucan_md = pd.DataFrame(ucan_md_dict)
print(ucan_md.shape)
#(8325, 29)
#(8275, 29)
ucan_md.head(2)

(8267, 37)


dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                                     source_dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   

                            patient_dir  \
0  ASPTCTX0001_npr258071002437-20160805   
1  ASPTCTX0001_npr258071002437-20160805   

                                                                       image_dir  \
0  CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                               dicom_img  \
0  1.2.840.113619.2.55.3.4240671514.785.1470288679.289.1   
1        1.2.840.113619.2.5.1207072.11651.1470306878.205   

                       patient_id patient_age patient_sex patient_weight  \
0   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   
1   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   

  patient_size   rows columns num_slices          pixel_spacing  \
0       '1.56'   2291     888          1   [0.545455, 0.596847]   
1       '1.56'    512     512       1153   [0.976562, 0.976562]   

  slice_thickness                                  img_pos  \
0   '1250.090942'               [-265.000, 0.000, 350.000]   
1      '0.949219'   [-233.886719, -207.519531, 134.462494]   

                                                        img_orient  \
0   [1.000000, -0.000000, 0.000000, 0.000000, 0.000000, -1.000000]   
1     [1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]   

                                                        for_uid att_corr  \
0   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            
1   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            

  recons_method                                           image_type  \
0                               ['ORIGINAL', 'PRIMARY', 'LOCALIZER']   
1                 ['DERIVED', 'SECONDARY', 'REFORMATTED', 'AVERAGE']   

  aquisition_dt aquisition_time         study_desc          series_desc  \
0    '20160805'        '124149'   'PET FDG WB 1 h'   'CT SCOUT HEAD IN'   
1    '20160805'        '130733'   'PET FDG WB 1 h'          'WB VEN AX'   

                         protocol corr_img modality           manufacturer  \
0   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   
1   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   

  manufacturer_model radiopharmaceutical radiopharmaceutical_volume  \
0    'Discovery STE'                                                  
1    'Discovery STE'                                                  

  radiopharmaceutical_start_time radionuclide_total_dose  \
0                                                          
1                                                          

  radionuclide_half_life radionuclide_positron_fraction  \
0                                                         
1                                                         

  radiopharmaceutical_start_date_time  
0                                      
1

In [10]:
ucan_md.dtypes
ucan_md['imgsz_x'] = np.int64(ucan_md['rows'].str.replace("'","").replace(" ",""))
ucan_md['imgsz_y'] = np.int64(ucan_md['columns'])
ucan_md['num_slices'] = np.int64(ucan_md['num_slices'])
ucan_md['voxsz_x'] = ucan_md['pixel_spacing'].apply(lambda x: (np.float64(x.replace(" ","").replace("[","").replace("]","").split(',')[0])))
ucan_md['voxsz_y'] = ucan_md['pixel_spacing'].apply(lambda x: (np.float64(x.replace(" ","").replace("[","").replace("]","").split(',')[1])))
ucan_md['slice_thickness'] = np.float64(ucan_md['slice_thickness'].str.replace("'","").replace(" ",""))
ucan_md.head(2)

dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                                     source_dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   

                            patient_dir  \
0  ASPTCTX0001_npr258071002437-20160805   
1  ASPTCTX0001_npr258071002437-20160805   

                                                                       image_dir  \
0  CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                               dicom_img  \
0  1.2.840.113619.2.55.3.4240671514.785.1470288679.289.1   
1        1.2.840.113619.2.5.1207072.11651.1470306878.205   

                       patient_id patient_age patient_sex patient_weight  \
0   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   
1   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   

  patient_size   rows columns  num_slices          pixel_spacing  \
0       '1.56'   2291     888           1   [0.545455, 0.596847]   
1       '1.56'    512     512        1153   [0.976562, 0.976562]   

   slice_thickness                                  img_pos  \
0      1250.090942               [-265.000, 0.000, 350.000]   
1         0.949219   [-233.886719, -207.519531, 134.462494]   

                                                        img_orient  \
0   [1.000000, -0.000000, 0.000000, 0.000000, 0.000000, -1.000000]   
1     [1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]   

                                                        for_uid att_corr  \
0   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            
1   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            

  recons_method                                           image_type  \
0                               ['ORIGINAL', 'PRIMARY', 'LOCALIZER']   
1                 ['DERIVED', 'SECONDARY', 'REFORMATTED', 'AVERAGE']   

  aquisition_dt aquisition_time         study_desc          series_desc  \
0    '20160805'        '124149'   'PET FDG WB 1 h'   'CT SCOUT HEAD IN'   
1    '20160805'        '130733'   'PET FDG WB 1 h'          'WB VEN AX'   

                         protocol corr_img modality           manufacturer  \
0   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   
1   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   

  manufacturer_model radiopharmaceutical radiopharmaceutical_volume  \
0    'Discovery STE'                                                  
1    'Discovery STE'                                                  

  radiopharmaceutical_start_time radionuclide_total_dose  \
0                                                          
1                                                          

  radionuclide_half_life radionuclide_positron_fraction  \
0                                                         
1                                                         

  radiopharmaceutical_start_date_time  imgsz_x  imgsz_y   voxsz_x   voxsz_y  
0                                         2291      888  0.545455  0.596847  
1                                          512      512  0.976562  0.976562

In [11]:
ucan_md['image_size'] = ucan_md.apply(lambda x: (x.imgsz_x, x.imgsz_y, x.num_slices), axis=1)
ucan_md['voxel_size'] = ucan_md.apply(lambda x: (x.voxsz_x, x.voxsz_y, x.slice_thickness), axis=1)
ucan_md.head(2)

dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                                     source_dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   

                            patient_dir  \
0  ASPTCTX0001_npr258071002437-20160805   
1  ASPTCTX0001_npr258071002437-20160805   

                                                                       image_dir  \
0  CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                               dicom_img  \
0  1.2.840.113619.2.55.3.4240671514.785.1470288679.289.1   
1        1.2.840.113619.2.5.1207072.11651.1470306878.205   

                       patient_id patient_age patient_sex patient_weight  \
0   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   
1   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   

  patient_size   rows columns  num_slices          pixel_spacing  \
0       '1.56'   2291     888           1   [0.545455, 0.596847]   
1       '1.56'    512     512        1153   [0.976562, 0.976562]   

   slice_thickness                                  img_pos  \
0      1250.090942               [-265.000, 0.000, 350.000]   
1         0.949219   [-233.886719, -207.519531, 134.462494]   

                                                        img_orient  \
0   [1.000000, -0.000000, 0.000000, 0.000000, 0.000000, -1.000000]   
1     [1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]   

                                                        for_uid att_corr  \
0   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            
1   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            

  recons_method                                           image_type  \
0                               ['ORIGINAL', 'PRIMARY', 'LOCALIZER']   
1                 ['DERIVED', 'SECONDARY', 'REFORMATTED', 'AVERAGE']   

  aquisition_dt aquisition_time         study_desc          series_desc  \
0    '20160805'        '124149'   'PET FDG WB 1 h'   'CT SCOUT HEAD IN'   
1    '20160805'        '130733'   'PET FDG WB 1 h'          'WB VEN AX'   

                         protocol corr_img modality           manufacturer  \
0   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   
1   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   

  manufacturer_model radiopharmaceutical radiopharmaceutical_volume  \
0    'Discovery STE'                                                  
1    'Discovery STE'                                                  

  radiopharmaceutical_start_time radionuclide_total_dose  \
0                                                          
1                                                          

  radionuclide_half_life radionuclide_positron_fraction  \
0                                                         
1                                                         

  radiopharmaceutical_start_date_time  imgsz_x  imgsz_y   voxsz_x   voxsz_y  \
0                                         2291      888  0.545455  0.596847   
1                                          512      512  0.976562  0.976562   

         image_size                         voxel_size  
0    (2291, 888, 1)  (0.545455, 0.596847, 1250.090942)  
1  (512, 512, 1153)     (0.976562, 0.976562, 0.949219)

In [12]:
ucan_md.columns

Index(['dir', 'source_dir', 'patient_dir', 'image_dir', 'dicom_img',
       'patient_id', 'patient_age', 'patient_sex', 'patient_weight',
       'patient_size', 'rows', 'columns', 'num_slices', 'pixel_spacing',
       'slice_thickness', 'img_pos', 'img_orient', 'for_uid', 'att_corr',
       'recons_method', 'image_type', 'aquisition_dt', 'aquisition_time',
       'study_desc', 'series_desc', 'protocol', 'corr_img', 'modality',
       'manufacturer', 'manufacturer_model', 'radiopharmaceutical',
       'radiopharmaceutical_volume', 'radiopharmaceutical_start_time',
       'radionuclide_total_dose', 'radionuclide_half_life',
       'radionuclide_positron_fraction', 'radiopharmaceutical_start_date_time',
       'imgsz_x', 'imgsz_y', 'voxsz_x', 'voxsz_y', 'image_size', 'voxel_size'],
      dtype='object')

In [13]:
#order columns
new_col_lst = ['dir', 'source_dir', 'patient_dir', 'image_dir', 'dicom_img',
       'patient_id', 'patient_age', 'patient_sex', 'patient_weight',
       'patient_size', 'rows', 'columns', 'imgsz_x', 'imgsz_y', 'num_slices', 
       'voxsz_x', 'voxsz_y', 'slice_thickness',  'pixel_spacing',
       'image_size', 'voxel_size', 'img_pos', 'img_orient', 'for_uid', 'att_corr',
       'recons_method', 'image_type', 'aquisition_dt', 'aquisition_time', 'study_desc',
       'series_desc', 'protocol', 'corr_img', 'modality', 'manufacturer',
       'manufacturer_model', 'radiopharmaceutical', 'radiopharmaceutical_volume',
       'radiopharmaceutical_start_time', 'radionuclide_total_dose', 
       'radionuclide_half_life', 'radionuclide_positron_fraction', 
       'radiopharmaceutical_start_date_time'
       ]

ucan_md = ucan_md[new_col_lst]

In [14]:
print("Toal CT folders: ", ucan_md[ucan_md['image_dir'].str.contains('CT')].shape[0])
print("Toal PT folders: ", ucan_md[ucan_md['image_dir'].str.contains('PT')].shape[0])

Toal CT folders:  6634
Toal PT folders:  2708


In [15]:
ucan_md.to_excel(destination_path)

In [16]:
ucan_md.num_slices.min()

1

In [17]:
ucan_md[ucan_md['num_slices']<=100].to_excel(os.path.join(source_path, 'exams_with_less_than_100_slices.xlsx'))

lessthan100slices_summ = ucan_md[np.int64(ucan_md['num_slices'])<=100].groupby(['modality', 'slice_thickness', 'num_slices']).agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False).reset_index()
lessthan100slices_summ.to_excel(os.path.join(source_path, 'exams_with_less_than_100_slices_final.xlsx'))

print(lessthan100slices_summ.shape)
#(785, 4)
lessthan100slices_summ.head(10)

(782, 4)


modality  slice_thickness  num_slices  dicom_img
0     'CT'      1250.090942           1        308
1     'CT'      1250.090942           2         65
2     'CT'      1310.818237           2         20
3     'CT'      1100.636353           2         19
4     'CT'         1.250000          17          6
5     'CT'       957.545471           2          4
6     'CT'      1018.545471           2          4
7     'CT'      1100.727339           2          4
8     'CT'       900.090881           1          3
9     'CT'       988.636414           2          3

In [18]:
ucan_md_voxeldist = ucan_md.groupby(['modality', 'voxel_size']).agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False).reset_index()
ucan_md_voxeldist.to_excel(os.path.join(source_path, 'ucan_exams_md_voxeldist.xlsx'))
ucan_md_voxeldist.head(10)


modality                         voxel_size  dicom_img
0     'PT'          (1.953125, 1.953125, 2.8)        740
1     'PT'           (3.90625, 3.90625, 3.27)        738
2     'CT'        (0.976562, 0.976562, 0.625)        630
3     'PT'         (1.953125, 1.953125, 2.79)        527
4     'CT'  (0.545455, 0.596847, 1250.090942)        373
5     'PT'            (3.90625, 3.90625, 2.8)        355
6     'PT'           (3.90625, 3.90625, 2.79)        275
7     'CT'          (0.976562, 0.976562, 3.0)        220
8     'CT'         (0.703125, 0.703125, 1.25)        166
9     'CT'         (1.367188, 1.367188, 3.75)        156

In [19]:
ucan_md_imgszdist = ucan_md.groupby(['modality', 'image_size']).agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False).reset_index()
ucan_md_imgszdist.to_excel(os.path.join(source_path, 'ucan_exams_md_imgszdist.xlsx'))
ucan_md_imgszdist.head(10)

modality        image_size  dicom_img
0     'PT'   (128, 128, 247)        478
1     'PT'   (256, 256, 314)        451
2     'PT'   (128, 128, 287)        318
3     'CT'    (2291, 888, 1)        308
4     'CT'   (512, 512, 322)        293
5     'PT'   (256, 256, 287)        239
6     'PT'   (128, 128, 314)        237
7     'CT'  (512, 512, 1151)        232
8     'CT'   (512, 512, 354)        230
9     'CT'   (512, 512, 644)        192

In [20]:
ucan_md_gt200slices = ucan_md[ucan_md['num_slices']>=200]
ucan_md_gt200slices.to_excel(os.path.join(source_path, 'ucan_exams_with_more_than_200_slices.xlsx'))
print(ucan_md_gt200slices.shape)
ucan_md_gt200slices.head(2)

(6522, 43)


dir  \
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   
2   E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-130439-6.3-UAS-WB-FDG-3D-CT-VENFAS-LUNG-1.25-AX-1.250000mm   

                                                     source_dir  \
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
2  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   

                            patient_dir  \
1  ASPTCTX0001_npr258071002437-20160805   
2  ASPTCTX0001_npr258071002437-20160805   

                                                                 image_dir  \
1  CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   
2   CT-20160805-130439-6.3-UAS-WB-FDG-3D-CT-VENFAS-LUNG-1.25-AX-1.250000mm   

                                               dicom_img  \
1        1.2.840.113619.2.5.1207072.11651.1470306878.205   
2  1.2.840.113619.2.55.3.4240671514.785.1470288680.829.1   

                       patient_id patient_age patient_sex patient_weight  \
1   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   
2   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   

  patient_size  rows columns  imgsz_x  imgsz_y  num_slices   voxsz_x  \
1       '1.56'   512     512      512      512        1153  0.976562   
2       '1.56'   512     512      512      512         230  0.976562   

    voxsz_y  slice_thickness          pixel_spacing        image_size  \
1  0.976562         0.949219   [0.976562, 0.976562]  (512, 512, 1153)   
2  0.976562         1.250000   [0.976562, 0.976562]   (512, 512, 230)   

                       voxel_size                                  img_pos  \
1  (0.976562, 0.976562, 0.949219)   [-233.886719, -207.519531, 134.462494]   
2      (0.976562, 0.976562, 1.25)             [-250.000, -250.000, 56.500]   

                                                      img_orient  \
1   [1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]   
2   [1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]   

                                                        for_uid att_corr  \
1   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            
2   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            

  recons_method                                           image_type  \
1                 ['DERIVED', 'SECONDARY', 'REFORMATTED', 'AVERAGE']   
2                                   ['ORIGINAL', 'PRIMARY', 'AXIAL']   

  aquisition_dt aquisition_time         study_desc      series_desc  \
1    '20160805'        '130733'   'PET FDG WB 1 h'      'WB VEN AX'   
2    '20160805'        '130733'   'PET FDG WB 1 h'   'LUNG 1.25 AX'   

                         protocol corr_img modality           manufacturer  \
1   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   
2   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   

  manufacturer_model radiopharmaceutical radiopharmaceutical_volume  \
1    'Discovery STE'                                                  
2    'Discovery STE'                                                  

  radiopharmaceutical_start_time radionuclide_total_dose  \
1                                                          
2                                                          

  radionuclide_half_life radionuclide_positron_fraction  \
1                                                         
2                                                         

  radiopharmaceutical_start_date_time  
1                                      
2

In [21]:
ucan_md_gt200slices.groupby('voxel_size').agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False)

dicom_img
voxel_size                               
(1.953125, 1.953125, 2.8)             736
(3.90625, 3.90625, 3.27)              732
(0.976562, 0.976562, 0.625)           630
(1.953125, 1.953125, 2.79)            519
(3.90625, 3.90625, 2.8)               353
...                                   ...
(1.056641, 1.056641, 0.625)             1
(1.054688, 1.054688, 3.75)              1
(0.746094, 0.746094, 0.746094)          1
(0.746094, 0.746094, 1.0)               1
(1.564941, 1.564941, 0.777344)          1

[923 rows x 1 columns]

In [22]:
ucan_md_gt200slices_voxeldist = ucan_md_gt200slices.groupby(['modality', 'voxel_size']).agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False).reset_index()
ucan_md_gt200slices_voxeldist.to_excel(os.path.join(source_path, 'ucan_exams_with_more_than_200_slices_voxeldist.xlsx'))
ucan_md_gt200slices_voxeldist.head(10)


modality                   voxel_size  dicom_img
0     'PT'    (1.953125, 1.953125, 2.8)        736
1     'PT'     (3.90625, 3.90625, 3.27)        732
2     'CT'  (0.976562, 0.976562, 0.625)        630
3     'PT'   (1.953125, 1.953125, 2.79)        519
4     'PT'      (3.90625, 3.90625, 2.8)        353
5     'PT'     (3.90625, 3.90625, 2.79)        272
6     'CT'    (0.976562, 0.976562, 3.0)        217
7     'CT'   (0.703125, 0.703125, 1.25)        158
8     'CT'   (1.367188, 1.367188, 3.75)        153
9     'CT'   (0.976562, 0.976562, 3.75)         85

In [23]:
ucan_md_gt200slices_imgszdist = ucan_md_gt200slices.groupby(['modality', 'image_size']).agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False).reset_index()
ucan_md_gt200slices_imgszdist.to_excel(os.path.join(source_path, 'ucan_exams_with_more_than_200_slices_imagesizedist.xlsx'))
ucan_md_gt200slices_imgszdist.head(10)

modality        image_size  dicom_img
0     'PT'   (128, 128, 247)        478
1     'PT'   (256, 256, 314)        451
2     'PT'   (128, 128, 287)        318
3     'CT'   (512, 512, 322)        293
4     'PT'   (256, 256, 287)        239
5     'PT'   (128, 128, 314)        237
6     'CT'  (512, 512, 1151)        232
7     'CT'   (512, 512, 354)        230
8     'CT'   (512, 512, 644)        192
9     'CT'   (512, 512, 382)        184

In [24]:
Lessthan_200Slices = ucan_md[ucan_md['num_slices']<200]
Lessthan_200Slices.to_excel(os.path.join(source_path, "less_than_200_slices.xlsx"))
print(Lessthan_200Slices.shape)
Lessthan_200Slices.head(2)

(1745, 43)


dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
7                    E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr462153436708-20220302/CT-20220302-130340-3.8-WB-F18-FDG-VENFAS-SCOUT-957.727295mm   

                                                     source_dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
7  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   

                            patient_dir  \
0  ASPTCTX0001_npr258071002437-20160805   
7  ASPTCTX0001_npr462153436708-20220302   

                                                                       image_dir  \
0  CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
7                    CT-20220302-130340-3.8-WB-F18-FDG-VENFAS-SCOUT-957.727295mm   

                                               dicom_img  \
0  1.2.840.113619.2.55.3.4240671514.785.1470288679.289.1   
7  1.2.840.113619.2.405.3.3842664208.13.1646146733.689.2   

                       patient_id patient_age patient_sex patient_weight  \
0   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   
7   'ASPTCTX0001_npr462153436708'      '076Y'         'O'         '75.0'   

  patient_size   rows columns  imgsz_x  imgsz_y  num_slices   voxsz_x  \
0       '1.56'   2291     888     2291      888           1  0.545455   
7       '1.68'   1755     835     1755      835           2  0.545455   

    voxsz_y  slice_thickness          pixel_spacing      image_size  \
0  0.596847      1250.090942   [0.545455, 0.596847]  (2291, 888, 1)   
7  0.634731       957.727295   [0.545455, 0.634731]  (1755, 835, 2)   

                          voxel_size                      img_pos  \
0  (0.545455, 0.596847, 1250.090942)   [-265.000, 0.000, 350.000]   
7   (0.545455, 0.634731, 957.727295)     [0.000, 265.000, 10.000]   

                                                        img_orient  \
0   [1.000000, -0.000000, 0.000000, 0.000000, 0.000000, -1.000000]   
7   [0.000000, -1.000000, 0.000000, 0.000000, 0.000000, -1.000000]   

                                                        for_uid att_corr  \
0   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1            
7    1.2.840.113619.2.405.3.3842664208.13.1646146733.682.6715.1            

  recons_method                             image_type aquisition_dt  \
0                 ['ORIGINAL', 'PRIMARY', 'LOCALIZER']    '20160805'   
7                 ['ORIGINAL', 'PRIMARY', 'LOCALIZER']    '20220302'   

    aquisition_time         study_desc          series_desc  \
0          '124149'   'PET FDG WB 1 h'   'CT SCOUT HEAD IN'   
7   '130433.876454'   'PET FDG WB 1 h'              'SCOUT'   

                         protocol corr_img modality           manufacturer  \
0   '6.3 UAS-WB-FDG-3D CT VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   
7         '3.8 WB F18 FDG VENFAS'              'CT'   'GE MEDICAL SYSTEMS'   

  manufacturer_model radiopharmaceutical radiopharmaceutical_volume  \
0    'Discovery STE'                                                  
7     'Discovery MI'                                                  

  radiopharmaceutical_start_time radionuclide_total_dose  \
0                                                          
7                                                          

  radionuclide_half_life radionuclide_positron_fraction  \
0                                                         
7                                                         

  radiopharmaceutical_start_date_time  
0                                      
7

In [25]:
ucan_md[ucan_md['image_dir'].str.contains('-ax-')][ucan_md['num_slices']<200][ucan_md['image_dir'].str.contains('PT-')]

C:\Users\asch4890\AppData\Local\Temp\ipykernel_21108\1214850953.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  ucan_md[ucan_md['image_dir'].str.contains('-ax-')][ucan_md['num_slices']<200][ucan_md['image_dir'].str.contains('PT-')]


Empty DataFrame
Columns: [dir, source_dir, patient_dir, image_dir, dicom_img, patient_id, patient_age, patient_sex, patient_weight, patient_size, rows, columns, imgsz_x, imgsz_y, num_slices, voxsz_x, voxsz_y, slice_thickness, pixel_spacing, image_size, voxel_size, img_pos, img_orient, for_uid, att_corr, recons_method, image_type, aquisition_dt, aquisition_time, study_desc, series_desc, protocol, corr_img, modality, manufacturer, manufacturer_model, radiopharmaceutical, radiopharmaceutical_volume, radiopharmaceutical_start_time, radionuclide_total_dose, radionuclide_half_life, radionuclide_positron_fraction, radiopharmaceutical_start_date_time]
Index: []

In [26]:
ucan_md[ucan_md['num_slices']<200]

dir  \
0               E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
7                                 E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr462153436708-20220302/CT-20220302-130340-3.8-WB-F18-FDG-VENFAS-SCOUT-957.727295mm   
8                                 E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr462153436708-20220302/CT-20220302-130340-3.8-WB-F18-FDG-VENFAS-SCOUT-966.272766mm   
19              E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr564116620440-20170515/CT-20170515-113049-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
26    E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr696287093420-20180814/CT-20180814-140104-6.5-UAS-WB-FDG-3D-CT-THORAX-BUK-NATIV-CT-SCOUT-HEAD-IN-1250.090942mm   
...                                                                                                                                                                                         ...   
8225                      E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr272766092791-20210825/CT-20210825-094704-3.8-WB-F18-FDG-VENFAS-SCOUT-1081.272705mm   
8236                          E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr272766092791-20201218/CT-20201218-090857-3.8-WB_FDG-VENFAS-SCOUT-1075.636353mm   
8237                          E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr272766092791-20201218/CT-20201218-090857-3.8-WB_FDG-VENFAS-SCOUT-1079.272705mm   
8248                          E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr272766092791-20200918/CT-20200918-100432-3.7-WB_FDG-VENFAS-SCOUT-1156.909058mm   
8249                          E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr272766092791-20200918/CT-20200918-100432-3.7-WB_FDG-VENFAS-SCOUT-1257.090942mm   

                                                               source_dir  \
0            E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
7            E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
8            E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
19           E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
26           E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
...                                                                   ...   
8225  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   
8236  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   
8237  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   
8248  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   
8249  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   

                               patient_dir  \
0     ASPTCTX0001_npr258071002437-20160805   
7     ASPTCTX0001_npr462153436708-20220302   
8     ASPTCTX0001_npr462153436708-20220302   
19    ASPTCTX0001_npr564116620440-20170515   
26    ASPTCTX0001_npr696287093420-20180814   
...                                    ...   
8225  ASPTCTX0001_npr272766092791-20210825   
8236  ASPTCTX0001_npr272766092791-20201218   
8237  ASPTCTX0001_npr272766092791-20201218   
8248  ASPTCTX0001_npr272766092791-20200918   
8249  ASPTCTX0001_npr272766092791-20200918   

                                                                                    image_dir  \
0               CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
7                                 CT-20220302-130340-3.8-WB-F18-FDG-VENFAS-SCOUT-957.727295mm   
8                                 CT-20220302-130340-3.8-WB-F18-FDG-VENFAS-SCO

### Match metadata with selected files

In [27]:
metadata = pd.read_excel(destination_path)
# metadata['dir'] = metadata['dir'].str.replace('/','\\')
print(metadata.shape)
metadata.head(2)

(8267, 44)


Unnamed: 0  \
0           0   
1           1   

                                                                                                                                                                               dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr258071002437-20160805/CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                                     source_dir  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   

                            patient_dir  \
0  ASPTCTX0001_npr258071002437-20160805   
1  ASPTCTX0001_npr258071002437-20160805   

                                                                       image_dir  \
0  CT-20160805-124130-6.3-UAS-WB-FDG-3D-CT-VENFAS-CT-SCOUT-HEAD-IN-1250.090942mm   
1        CT-20160805-131025-6.3-UAS-WB-FDG-3D-CT-VENFAS-WB-VEN-AX-0.949219mm_JK_   

                                               dicom_img  \
0  1.2.840.113619.2.55.3.4240671514.785.1470288679.289.1   
1        1.2.840.113619.2.5.1207072.11651.1470306878.205   

                       patient_id patient_age patient_sex patient_weight  \
0   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   
1   'ASPTCTX0001_npr258071002437'      '070Y'         'O'         '46.0'   

  patient_size  rows  columns  imgsz_x  imgsz_y  num_slices   voxsz_x  \
0       '1.56'  2291      888     2291      888           1  0.545455   
1       '1.56'   512      512      512      512        1153  0.976562   

    voxsz_y  slice_thickness          pixel_spacing        image_size  \
0  0.596847      1250.090942   [0.545455, 0.596847]    (2291, 888, 1)   
1  0.976562         0.949219   [0.976562, 0.976562]  (512, 512, 1153)   

                          voxel_size                                  img_pos  \
0  (0.545455, 0.596847, 1250.090942)               [-265.000, 0.000, 350.000]   
1     (0.976562, 0.976562, 0.949219)   [-233.886719, -207.519531, 134.462494]   

                                                        img_orient  \
0   [1.000000, -0.000000, 0.000000, 0.000000, 0.000000, -1.000000]   
1     [1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]   

                                                        for_uid att_corr  \
0   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1      NaN   
1   1.2.840.113619.2.55.3.4240671514.785.1470288679.287.12920.1      NaN   

  recons_method                                           image_type  \
0           NaN                 ['ORIGINAL', 'PRIMARY', 'LOCALIZER']   
1           NaN   ['DERIVED', 'SECONDARY', 'REFORMATTED', 'AVERAGE']   

  aquisition_dt aquisition_time         study_desc          series_desc  \
0    '20160805'        '124149'   'PET FDG WB 1 h'   'CT SCOUT HEAD IN'   
1    '20160805'        '130733'   'PET FDG WB 1 h'          'WB VEN AX'   

                         protocol corr_img modality           manufacturer  \
0   '6.3 UAS-WB-FDG-3D CT VENFAS'      NaN     'CT'   'GE MEDICAL SYSTEMS'   
1   '6.3 UAS-WB-FDG-3D CT VENFAS'      NaN     'CT'   'GE MEDICAL SYSTEMS'   

  manufacturer_model radiopharmaceutical radiopharmaceutical_volume  \
0    'Discovery STE'                 NaN                        NaN   
1    'Discovery STE'                 NaN                        NaN   

  radiopharmaceutical_start_time radionuclide_total_dose  \
0                            NaN                     NaN   
1                            NaN                     NaN   

  radionuclide_half_life radionuclide_positron_fraction  \
0                    NaN                            NaN   
1                    NaN                            NaN   

  radiopharmaceutical_start_date_t

In [32]:
selected_imgs = pd.read_excel(selection_dataframe)
print(selected_imgs.shape)
selected_imgs.head(2)

(1316, 8)


Unnamed: 0  \
0           0   
1           1   

                                                                                                                                                                                       directory  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr748236341307-20220908/PT-20220908-103801-3.11-WB-F18-FDG-VENFAS--THORAX-INANDAD-QCFX-S-400-M.Free-2.8000mm   
1   E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr748236341307-20220908/CT-20220908-105049-3.11-WB-F18-FDG-VENFAS--THORAX-INANDAD-WB-Venfas-3-ax-3.000000mm   

                                                      source_directory  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   

                      patient_directory  \
0  ASPTCTX0001_npr748236341307-20220908   
1  ASPTCTX0001_npr748236341307-20220908   

                                                                            PET-CT_info  \
0  PT-20220908-103801-3.11-WB-F18-FDG-VENFAS--THORAX-INANDAD-QCFX-S-400-M.Free-2.8000mm   
1   CT-20220908-105049-3.11-WB-F18-FDG-VENFAS--THORAX-INANDAD-WB-Venfas-3-ax-3.000000mm   

        system              npr scan_date  
0  ASPTCTX0001  npr748236341307  20220908  
1  ASPTCTX0001  npr748236341307  20220908

In [33]:
master_data = pd.merge(selected_imgs, metadata, how="inner", left_on=['directory'], right_on=['dir'], sort=True, suffixes=("_x", "_y"))

col_drop_lst = ['dir', 'source_dir', 'patient_dir',	'image_dir', 'patient_sex', 'rows',	'columns', 'pixel_spacing']

master_data = master_data.drop(columns=col_drop_lst)

master_data['patient_id'] = master_data['patient_id'].apply(lambda x: x.split('_')[1].replace("'",""))
master_data['patient_age'] = master_data['patient_age'].apply(lambda x: x.replace("'","").replace("Y","").replace(" ",""))#.astype('Int64')
master_data['modality'] = master_data['modality'].apply(lambda x: x.strip().replace("'","").strip())

master_data.to_excel(os.path.join(source_path, "Finalized_dataset.xlsx"))

print(master_data.shape)
master_data.head(2)

(1316, 44)


Unnamed: 0_x  \
0           464   
1           465   

                                                                                                                                                               directory  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr134758508773-20191212/CT-20191212-111939-3.7-WB_FDG-CT-VENFAS-WB-Venfas-0.6-ax-0.625000mm   
1      E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB/ASPTCTX0001_npr134758508773-20191212/PT-20191212-110507-3.7-WB_FDG-CT-VENFAS-QCFX-S-400-2.7900mm_JK_   

                                               source_directory  \
0  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   
1  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221205__06__n86_90GB   

                      patient_directory  \
0  ASPTCTX0001_npr134758508773-20191212   
1  ASPTCTX0001_npr134758508773-20191212   

                                                           PET-CT_info  \
0  CT-20191212-111939-3.7-WB_FDG-CT-VENFAS-WB-Venfas-0.6-ax-0.625000mm   
1      PT-20191212-110507-3.7-WB_FDG-CT-VENFAS-QCFX-S-400-2.7900mm_JK_   

        system              npr scan_date  Unnamed: 0_y  \
0  ASPTCTX0001  npr134758508773  20191212           164   
1  ASPTCTX0001  npr134758508773  20191212           168   

                                               dicom_img       patient_id  \
0  1.2.840.113619.2.405.3.3842664208.103.1576074305.16.1  npr134758508773   
1      1.2.840.113619.2.405.3842664208.1576145387.106224  npr134758508773   

  patient_age patient_weight patient_size  imgsz_x  imgsz_y  num_slices  \
0         074         '72.0'       '1.64'      512      512        1763   
1         074         '72.0'       '1.64'      256      256         395   

    voxsz_x   voxsz_y  slice_thickness        image_size  \
0  0.976562  0.976562            0.625  (512, 512, 1763)   
1  1.953125  1.953125            2.790   (256, 256, 395)   

                    voxel_size  \
0  (0.976562, 0.976562, 0.625)   
1   (1.953125, 1.953125, 2.79)   

                                                   img_pos  \
0                             [-250.000, -250.000, 15.060]   
1   [-249.02343750000, -249.02343750000, -1082.9699707031]   

                                                      img_orient  \
0   [1.000000, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]   
1                                            [1, 0, 0, 0, 1, -0]   

                                                        for_uid  \
0   1.2.840.113619.2.405.3.3842664208.103.1576074304.305.5240.1   
1   1.2.840.113619.2.405.3.3842664208.103.1576074304.305.5240.1   

                                          att_corr recons_method  \
0                                              NaN           NaN   
1   'measured,, 0.000000 cm-1, attenuation smooth'        'QCFX'   

                          image_type aquisition_dt   aquisition_time  \
0   ['ORIGINAL', 'PRIMARY', 'AXIAL']    '20191212'   '112115.942112'   
1            ['ORIGINAL', 'PRIMARY']    '20191212'          '110507'   

          study_desc          series_desc                 protocol  \
0   'PET FDG WB 1 h'   'WB Venfas 0.6 ax'   '3.7 WB_FDG CT VENFAS'   
1   'PET FDG WB 1 h'         'QCFX-S 400'   '3.7 WB_FDG CT VENFAS'   

                                                                corr_img  \
0                                                                    NaN   
1   ['DECY', 'ATTN', 'SCAT', 'DTIM', 'RANSNG', 'DCAL', 'SLSENS', 'NORM']   

  modality           manufacturer manufacturer_model  \
0       CT   'GE MEDICAL SYSTEMS'     'Discovery MI'   
1       PT   'GE MEDICAL SYSTEMS'     'Discovery MI'   

            radiopharmaceutical radiopharmaceutical_volume  \
0                           NaN                        NaN   
1   'FDG -- fluorodeoxyglucose'                      '0.0'   

  radiopharmaceutical_start_time radionuclide_total_dose  \
0                            NaN                     NaN   
1                    

In [34]:
print(master_data[master_data['patient_age']==''].shape)

master_data[master_data['patient_age']=='']

(4, 44)


Unnamed: 0_x  \
741           1088   
779            762   
1135          1091   
1309           677   

                                                                                                                                                                                    directory  \
741                              E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221211__14_n574_1.2TBondisc/ASPTCTX0001_npr791408621040-20160525/PT-20160525-131734-UAS-WB-FDG-3D-WB-FDG-AC-3.2700mm   
779               E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221211__14_n574_1.2TBondisc/ASPTCTX0001_npr833769881976-20180414/PT-20180414-150311-6.39-WB_FDG-CT-VENFAS-WB-AC-FDG-VPFX-S-2.7900mm   
1135  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc/ASPTCTX0001_npr671684118786-20160518/PT-20160518-133807-UAS-WB-FDG-3D-VENFASTHORAX-INSP-20160120-WB-FDG-AC-3.2700mm   
1309                 E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221227_until_kl1215_n50_100GB/ASPTCTX0001_npr735028517794-20180928/PT-20180928-103145-UAS-WB-FDG-3D-CT-VENFAS-WB-FDG-AC-3.2700mm   

                                                           source_directory  \
741     E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221211__14_n574_1.2TBondisc   
779     E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221211__14_n574_1.2TBondisc   
1135    E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221215__17_n199_424GBondisc   
1309  E:/U-CAN-Lymfom_A/Raw_DCM_transf_date_20221227_until_kl1215_n50_100GB   

                         patient_directory  \
741   ASPTCTX0001_npr791408621040-20160525   
779   ASPTCTX0001_npr833769881976-20180414   
1135  ASPTCTX0001_npr671684118786-20160518   
1309  ASPTCTX0001_npr735028517794-20180928   

                                                                         PET-CT_info  \
741                              PT-20160525-131734-UAS-WB-FDG-3D-WB-FDG-AC-3.2700mm   
779               PT-20180414-150311-6.39-WB_FDG-CT-VENFAS-WB-AC-FDG-VPFX-S-2.7900mm   
1135  PT-20160518-133807-UAS-WB-FDG-3D-VENFASTHORAX-INSP-20160120-WB-FDG-AC-3.2700mm   
1309                   PT-20180928-103145-UAS-WB-FDG-3D-CT-VENFAS-WB-FDG-AC-3.2700mm   

           system              npr scan_date  Unnamed: 0_y  \
741   ASPTCTX0001  npr791408621040  20160525          2372   
779   ASPTCTX0001  npr833769881976  20180414          1932   
1135  ASPTCTX0001  npr671684118786  20160518          6625   
1309  ASPTCTX0001  npr735028517794  20180928           543   

                                              dicom_img       patient_id  \
741   1.2.840.113619.2.131.4240671514.1464175398.156805  npr791408621040   
779   1.2.840.113619.2.405.2831173121.1523711206.107827  npr833769881976   
1135  1.2.840.113619.2.131.4240671514.1463571833.840715  npr671684118786   
1309  1.2.840.113619.2.131.4240671514.1538123862.293064  npr735028517794   

     patient_age patient_weight patient_size  imgsz_x  imgsz_y  num_slices  \
741                      '65.0'       '1.64'      128      128         247   
779                      '66.0'       '1.73'      256      256         287   
1135                     '49.0'       '1.52'      128      128         248   
1309                     '98.0'       '1.65'      128      128         247   

       voxsz_x   voxsz_y  slice_thickness       image_size  \
741   3.906250  3.906250             3.27  (128, 128, 247)   
779   1.953125  1.953125             2.79  (256, 256, 287)   
1135  3.906250  3.906250             3.27  (128, 128, 248)   
1309  3.906250  3.906250             3.27  (128, 128, 247)   

                      voxel_size  \
741     (3.90625, 3.90625, 3.27)   
779   (1.953125, 1.953125, 2.79)   
1135    (3.90625, 3.90625, 3.27)   
1309    (3.90625, 3.90625, 3.27)   

                                                      img_pos  \
741    [-248.04687500000, -248.04687500000, -653.40002441406]   
779    [-249.02343750000, -249.02343750000, -563.25000000000]   
1135   [-248.04687500000, -248.04687500000, -695.40002441406]   
1309   [-248.04687500000, 

In [35]:
master_data_imgszdist = master_data.groupby(['modality', 'image_size']).agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False).reset_index()
master_data_imgszdist.to_excel(os.path.join(source_path, "master_data_imgszdist.xlsx"))
print(master_data_imgszdist.shape)
master_data_imgszdist.head(10)

(115, 3)


modality        image_size  dicom_img
0       PT   (128, 128, 247)        163
1       PT   (256, 256, 287)        105
2       PT   (256, 256, 314)         98
3       CT   (512, 512, 322)         93
4       CT  (512, 512, 1151)         90
5       CT   (512, 512, 354)         88
6       PT   (256, 256, 341)         65
7       CT   (512, 512, 382)         62
8       PT   (128, 128, 287)         59
9       CT   (512, 512, 215)         57

In [36]:
master_data_voxszdist = master_data.groupby(['modality', 'voxel_size']).agg({'dicom_img':'count'}).sort_values('dicom_img' ,ascending=False).reset_index()
master_data_voxszdist.to_excel(os.path.join(source_path, "master_data_voxszdist.xlsx"))
print(master_data_voxszdist.shape)
master_data_voxszdist.head(10)

(200, 3)


modality                      voxel_size  dicom_img
0       PT        (3.90625, 3.90625, 3.27)        280
1       PT      (1.953125, 1.953125, 2.79)        201
2       PT       (1.953125, 1.953125, 2.8)        154
3       CT       (0.976562, 0.976562, 3.0)        142
4       CT      (0.976562, 0.976562, 3.75)         71
5       CT     (0.976562, 0.976562, 0.625)         40
6       CT  (0.976562, 0.976562, 0.949219)         38
7       PT        (5.46875, 5.46875, 3.27)         23
8       CT      (1.367188, 1.367188, 3.75)         21
9       CT  (0.703125, 0.703125, 0.949219)         13

In [37]:
master_data.columns

Index(['Unnamed: 0_x', 'directory', 'source_directory', 'patient_directory',
       'PET-CT_info', 'system', 'npr', 'scan_date', 'Unnamed: 0_y',
       'dicom_img', 'patient_id', 'patient_age', 'patient_weight',
       'patient_size', 'imgsz_x', 'imgsz_y', 'num_slices', 'voxsz_x',
       'voxsz_y', 'slice_thickness', 'image_size', 'voxel_size', 'img_pos',
       'img_orient', 'for_uid', 'att_corr', 'recons_method', 'image_type',
       'aquisition_dt', 'aquisition_time', 'study_desc', 'series_desc',
       'protocol', 'corr_img', 'modality', 'manufacturer',
       'manufacturer_model', 'radiopharmaceutical',
       'radiopharmaceutical_volume', 'radiopharmaceutical_start_time',
       'radionuclide_total_dose', 'radionuclide_half_life',
       'radionuclide_positron_fraction',
       'radiopharmaceutical_start_date_time'],
      dtype='object')

In [38]:
master_data = pd.read_excel(os.path.join(source_path, "Finalized_dataset.xlsx"))

#master_data['modality'] = master_data['modality'].apply(lambda x: x.strip().replace("'","").strip())
#master_data.to_csv('F:/U-CAN-Lymfom_A/Selected_for_UCAN_project/Selected_exams_from_U-CAN-Lymfom_A_MasterDataset_2Oct2023.csv', index=False)


In [39]:
print("Mode of x image dims: ", master_data['imgsz_x'].mode()[0])
print("Mode of y image dims: ", master_data['imgsz_y'].mode()[0])
print("Mode of z image dims: ", master_data['num_slices'].mode()[0])
print("Mode of x voxel dims: ", master_data['voxsz_x'].mode()[0])
print("Mode of y voxel dims: ", master_data['voxsz_y'].mode()[0])
print("Mode of z voxel dims: ", master_data['slice_thickness'].mode()[0])

Mode of x image dims:  512
Mode of y image dims:  512
Mode of z image dims:  287
Mode of x voxel dims:  1.953125
Mode of y voxel dims:  1.953125
Mode of z voxel dims:  3.0


In [40]:
master_data[['imgsz_x','imgsz_y','num_slices','voxsz_x','voxsz_y','slice_thickness']].describe()

imgsz_x      imgsz_y   num_slices      voxsz_x      voxsz_y  \
count  1316.000000  1316.000000  1316.000000  1316.000000  1316.000000   
mean    354.528875   354.528875   462.604103     1.927715     1.927715   
std     163.868147   163.868147   360.600113     1.241188     1.241188   
min     128.000000   128.000000    89.000000     0.703125     0.703125   
25%     256.000000   256.000000   269.000000     0.976562     0.976562   
50%     384.000000   384.000000   314.000000     1.889648     1.889648   
75%     512.000000   512.000000   382.000000     1.953125     1.953125   
max     512.000000   512.000000  1763.000000     5.468750     5.468750   

       slice_thickness  
count      1316.000000  
mean          2.679454  
std           0.885059  
min           0.625000  
25%           2.790000  
50%           3.000000  
75%           3.270000  
max           3.750000

### New size and spacing based on master data

new_size = [384, 384, 384]
new_spacing = [2, 2, 3]

In [41]:
master_data.head(2)
master_data.npr.nunique()

457

In [42]:
master_data.patient_directory.nunique()

658